frontend2/src/components/elements/Button.tsx

```diff
@@ -36,7 +44,15 @@ const Button: React.FC<ButtonProps> = ({
       {...rest}
     >
       {iconName !== undefined && <Icon name={iconName} size="sm" />}
-      {label}
+      {loading ? (
+        <Spinner
+          className="mx-2"
+          variant={variant.includes("danger") ? "danger" : ""}
```

Comment:
Ahh I think this could be another use of variant typing... not sure this `.includes(...)` is very safe?

Reply:
I'll change it to === checks